packages/hash/api/src/model/knowledge/link.model.ts

```diff
@@ -0,0 +1,160 @@
+    linkTypeModel,
+    targetEntity,
+  }: LinkModelConstructorParams) {
+    this.accountId = accountId;
+
+    this.sourceEntity = sourceEntity;
+    this.linkTypeModel = linkTypeModel;
+    this.targetEntity = targetEntity;
+  }
+
+  /**
+   * Create a link between a source and a target entity using a specific link
+   * type.
+   *
+   * @param params.accountId the accountId of the account creating the link
+   * @param params.sourceEntity the source entity of the link
+   * @param params.linkTypeModel the Link Type of the link
+   * @param params.targetEntity the target entity of the link
+   */
+  static async create(
+    graphApi: GraphApi,
+    {
+      accountId,
+      sourceEntity,
+      linkTypeModel,
+      targetEntity,
+    }: LinkModelCreateParams,
+  ): Promise<LinkModel> {
+    const {
+      data: { sourceEntityId, linkTypeUri, targetEntityId },
+    } = await graphApi.createLink(sourceEntity.entityId, {
+      accountId,
+      linkTypeUri: linkTypeModel.schema.$id,
+      targetEntityId: targetEntity.entityId,
+    });
+
+    const fetchedParams = {
+      sourceEntity: await EntityModel.getLatest(graphApi, {
+        accountId,
+        entityId: sourceEntityId,
+      }),
+      linkTypeModel: await LinkTypeModel.get(graphApi, {
+        versionedUri: linkTypeUri,
+      }),
+      targetEntity: await EntityModel.getLatest(graphApi, {
+        accountId,
+        entityId: targetEntityId,
+      }),
+    };
+
+    return new LinkModel({
+      accountId,
+      ...fetchedParams,
+    });
+  }
+
+  /**
+   * Get all outgoing links of a source entity.
+   *
+   * @param params.accountId the accountId of the account creating the link
+   * @param params.sourceEntity the source entity of the link
+   */
+  static async getAllOutgoing(
+    graphApi: GraphApi,
+    { sourceEntity }: { sourceEntity: EntityModel },
+  ): Promise<LinkModel[]> {
+    const { data: entityLinks } = await graphApi.getEntityLinks(
+      sourceEntity.entityId,
+    );
+    return Promise.all(
+      Object.entries(entityLinks.outgoing)
+        /**
+         * When target entity id is a list the link has many targets.
+         * This is not supported yet
+         * @todo support 1:N links
+         * */
+        .filter(([_, targetEntityId]) => typeof targetEntityId === "string")
+        .map(
+          async ([linkTypeUri, targetEntityId]) =>
+            new LinkModel({
+              accountId: sourceEntity.accountId,
+              linkTypeModel: await LinkTypeModel.get(graphApi, {
+                versionedUri: linkTypeUri,
+              }),
+              sourceEntity,
+              targetEntity: await EntityModel.getLatest(graphApi, {
+                /** @todo figure out what account ID we use here */
+                accountId: sourceEntity.accountId,
+                entityId: targetEntityId,
+              }),
+            }),
+        ),
+    );
+  }
+
+  /**
+   * Get the outgoing link of a source entity given a link type.
```

Comment:
I think this is similar to the above but I don't see how this can be guaranteed to return just one thing. What if there are multiple `Friend` links going from `Bob`

Reply:
It isn't guaranteed, and could be an array of links of the same `Friend` type. We don't support inserting multiple types of the same type yet, though (the whole 1:N)

Reply:
Are we going to keep this method when we do support multiple links? Are we going to update this method and its docs (should we have a todo here then?)

Reply:
Todo added 1f4b2c774ee8da8af2f7abffa7d7356491fd1a72